lib/src/git.rs

```diff
@@ -2418,3 +2422,14 @@ pub struct Progress {
     pub bytes_downloaded: Option<u64>,
     pub overall: f32,
 }
+
+/// Allows temporarily overriding the behavior of a single `git fetch`
+/// operation as to whether tags are fetched
+#[derive(Copy, Clone, Debug)]
+pub enum FetchTagsOverride {
+    /// For this one fetch attempt, fetch all tags regardless of what the
+    /// remote's `tagOpt` is configured to
+    ForceAllTags,
+    /// No particular override, respect the remote's `tagOpt`.
+    UseRemoteConfiguration,
+}
```

Comment:
nit: maybe we can make it look similar to `gix::remote::fetch::Tags`, something like `None, All, Default`? We can add `FetchTags::None` if needed.

Reply:
I ended up reworking this so that `git_fetch` takes an `Option<FetchTagsOverride>` and the variants are now `AllTags` or `NoTags` (which map to `--tags` and `--no-tags` respectively). I avoided adding a `Default` since a) there's no git CLI argument that would map to, and b) I think having the outer `Option` makes it clearer that `None` means "nothing to override"

Let me know what you think of this version!